GpxExport/WorkoutDataStore.swift

```diff
@@ -67,9 +68,22 @@ class WorkoutDataStore {
                         }
                         return
                     }
-                    
-                    DispatchQueue.main.async {
-                        completion(locationResults, done, error)
+
+                    if done {
+                        allDone += 1
```

Comment:
as far as I understand, this is about HealthKit calling the completion handler with `done=true` and then contiuing to call it some more. I've seen this happen in the past with one specific run and I don't quite understand why it would do that.

Regardless. Are you sure that incrementing by 1 and then later compareing it to the sample count is ok?

Also, I would argue that a variable called `allDone` implies a boolean but it's actually a counter here, making this slightly confusing to a reader what about a better name?

Reply:
If there are more route samples, then completion is called with done=true at the end of every sample. Which leads to a closed file after the first sample has ended and a crash afterwards because there are attempts to write to the file for the other samples.

Reply:
My rides and runs are divided into several samples very often. I think it has to do with pausing a workout. Although it also seem to happen on rides where I stop somewhere for several minutes without having paused the workout.

You're right about the variable name. Have to think about a better one.

I did some additional tests. The last test resulted in a wrongly combined track. Have to investigate that further. Once I found the reason I will post the solution here.